src/session.cpp

```diff
@@ -287,6 +287,45 @@ namespace libtorrent
 	// configurations this will give a link error
 	void TORRENT_CFG() {}
 
+	session_params read_session_params(bdecode_node const& e, std::uint32_t const flags)
```

Comment:
this is the new free function you mentioned. I deprecated the member one, and it was a mess, it's used a lot